src/evaluate/utils/gradio.py

```diff
@@ -0,0 +1,123 @@
+import json
+import os
+import re
+import sys
+from pathlib import Path
+
+from datasets import Value
+
+from .logging import get_logger
+
+
+logger = get_logger(__name__)
+
+REGEX_YAML_BLOCK = re.compile(r"---[\n\r]+([\S\s]*?)[\n\r]+---[\n\r]")
+
+
+def infer_gradio_input_types(feature_types):
```

Comment:
I don't think you do this in the repo rn, but WDYT of making internal functions a bit more explicitly internal/private? That way other people don't import it and handling backwards compatibility is easier.

I would just add `_` prefix and don't export it. Same for other functions

Reply:
For this specifically the reason I think it would be a good idea is to expose them to the user is that some metrics might require a custom gradio widget and the user can easily reuse these helper functions to make it more useful.

Do you think that's not necessary or would you avoid custom widgets?

Reply:
I think this might be a bit too early of an optimization, but I Have no strong opinion, so feel free to make it public if you think it will be useful to users